script.extendedinfo/resources/lib/TheMovieDB.py

```diff
@@ -923,134 +941,154 @@ def extended_episode_info(tvshow_id, season, episode, cache_time=7):
-              "guest_stars": handle_people(response["credits"]["guest_stars"]),
               "videos": handle_videos(response["videos"]["results"]) if "videos" in response else [],
               "images": handle_images(response["images"]["stills"])}
     return (handle_episodes([response])[0], answer, response.get("account_states"))
 
 
 def extended_actor_info(actor_id):
+    '''
+    get ListItem and lists with extended info for actor with *actor_id
+    '''
     if not actor_id:
         return None
-    response = get_data(url="person/%s" % (actor_id),
+    data = get_data(url="person/%s" % (actor_id),
                         params={"append_to_response": ALL_ACTOR_PROPS},
                         cache_days=1)
-    if not response:
-        Utils.notify("Could not find actor info")
+    if not data:
+        utils.notify("Could not find actor info")
         return None
-    listitems = {"movie_roles": handle_movies(response["movie_credits"]["cast"]),
-                 "tvshow_roles": handle_tvshows(response["tv_credits"]["cast"]),
-                 "movie_crew_roles": handle_movies(response["movie_credits"]["crew"]),
-                 "tvshow_crew_roles": handle_tvshows(response["tv_credits"]["crew"]),
-                 "tagged_images": handle_images(response["tagged_images"]["results"]) if "tagged_images" in response else [],
-                 "images": handle_images(response["images"]["profiles"])}
-    info = Utils.ListItem(label=response['name'],
-                          path="%sextendedactorinfo&&id=%s" % (PLUGIN_BASE, response['id']),
-                          infos={'mediatype': "artist"})
-    info.set_properties({'adult': response.get('adult'),
-                         'alsoknownas': " / ".join(response.get('also_known_as', [])),
-                         'biography': Utils.clean_text(response.get('biography')),
-                         'birthday': response.get('birthday'),
-                         'age': Utils.calculate_age(response.get('birthday'), response.get('deathday')),
-                         'character': response.get('character'),
-                         'department': response.get('department'),
-                         'job': response.get('job'),
-                         'id': response['id'],
-                         'gender': genders.get(response['gender']),
-                         'cast_id': response.get('cast_id'),
-                         'credit_id': response.get('credit_id'),
-                         'deathday': response.get('deathday'),
-                         'placeofbirth': response.get('place_of_birth'),
-                         'homepage': response.get('homepage'),
-                         "DBMovies": len([d for d in listitems["movie_roles"] if "dbid" in d])})
-    info.set_artwork(get_image_urls(profile=response.get("profile_path")))
-    return (info, listitems)
+    lists = {"movie_roles": handle_movies(data["movie_credits"]["cast"]).reduce("character"),
```

Comment:
I had to read up on this to understand how this works. Interesting, learned something new. thanks